### PR TITLE
Homework Podstawy C++ vector-of-shared-ptrs

### PR DESCRIPTION
**Pull request by PawelYak**
Authors: @rgruberski, @PawelYak

### CODE REVIEW

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @rgruberski

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @PawelYak

**Comment by ziobron** [comment]
✅ CI Passed - Task: vector-of-shared-ptrs
🏅 Granted 5 XP for @rgruberski

**Comment by ziobron** [comment]
⏱ 3 XP extra for @rgruberski for on-time delivery

**Comment by ziobron** [comment]
✅ CI Passed - Task: vector-of-shared-ptrs
🏅 Granted 5 XP for @PawelYak

**Comment by ziobron** [comment]
⏱ 3 XP extra for @PawelYak for on-time delivery

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#include "vectorFunctions.hpp"
+#include <algorithm>
+#include <cstdio>
+#include <iostream>
+#include <memory>
+#include <vector>
+
+std::vector<std::shared_ptr<int>> generate(int count) {
+  std::vector<std::shared_ptr<int>> vec{};
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:9]: reserve memory for vec

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  for (auto i = 0; i < count; ++i) {
+    vec.push_back(std::make_shared<int>(i));
+  }
+  return vec;
+}
+
+void print(std::vector<std::shared_ptr<int>> vec) {
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:16]: Pass by const refrence. Try always avoid copy.

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  for (auto it : vec) {
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:17]: Pass by const refrence. Try always avoid copy.

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    std::cout << *it << '\n';
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:18]: ptr could be nullptr

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  }
+}
+
+void sub10(int *const el) {
+  if (el) {
+    *el -= 10;
+  }
+}
+
+void sub10(std::vector<std::shared_ptr<int>> vec) {
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:35]: Pass by const refrence. Try always avoid copy.

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**